src/renderer/port-forward/port-forward.store.ts

```diff
@@ -0,0 +1,218 @@
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+
+import { computed, IReactionDisposer, makeObservable, observable, reaction } from "mobx";
+import { ItemObject, ItemStore } from "../../common/item.store";
+import { autoBind, createStorage, getHostedClusterId, openExternal } from "../utils";
+import { PortForwardItem } from "./port-forward-item";
+import { apiBase } from "../api";
+import { waitUntilFree } from "tcp-port-used";
+import { Notifications } from "../components/notifications";
+import logger from "../../common/logger";
+
+export interface ForwardedPort {
+  clusterId?: string;
+  kind: string;
+  namespace: string;
+  name: string;
+  port: string;
+  forwardPort: string;
+}
+
+export class PortForwardStore extends ItemStore<PortForwardItem> {
+  private storage = createStorage<ForwardedPort[] | undefined>("port_forwards", undefined);
+
+  constructor() {
+    super();
+    makeObservable(this);
+    autoBind(this);
+
+    this.init();
+  }
+
+  private async init() {
+    await this.storage.whenReady;
+
+    const savedPortForwards = this.storage.get(); // undefined on first load
+
+    if (Array.isArray(savedPortForwards)) {
+      logger.info("[PORT_FORWARD] starting saved port-forwards");
+      await Promise.all(savedPortForwards.map(pf => {
+        const port = new PortForwardItem;
+
+        port.clusterId = pf.clusterId;
+        port.kind = pf.kind;
+        port.namespace = pf.namespace;
+        port.name = pf.name;
+        port.port = pf.port;
+        port.forwardPort = pf.forwardPort;
+
+        return addPortForward(port);
+      }));
+
+      this.portForwards = [];
+    }
+  }
+
+  @observable selectedItemId?: string;
+  @observable portForwards: PortForwardItem[];
+
+  @computed get selectedItem() {
+    return this.portForwards.find((e: ItemObject) => e.getId() === this.selectedItemId);
+  }
+
+  watch() {
+    const disposers: IReactionDisposer[] = [
+      reaction(() => this.portForwards, () => this.loadAll()),
+    ];
+
+    return () => disposers.forEach((dispose) => dispose());
+  }
+
+  loadAll() {
+    return this.loadItems(async () => {
+      let portForwards = await getPortForwards();
+
+      // filter out any not for this cluster
+      portForwards = portForwards?.filter(pf => pf.clusterId == getHostedClusterId());
```

Comment:
Shouldn't this be part of the `get6PortForwards` call itself?

Reply:
could be, but I envision a global page showing all port-forwards for all clusters

Reply:
I just think it is a bit strange for this component (which is built for in-cluster use) to get all the port forwards and then throw some away.